Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -325,35 +323,21 @@ public void stopWebApp(String sid, String appid) throws IOException {
     /**
      * List Web Apps by Subscription ID.
      */
+    @Deprecated
     public List<ResourceEx<WebApp>> listWebAppsOnWindowsBySubscriptionId(String sid, boolean force) {
-        if (!force && subscriptionIdToWebAppsOnWindowsMap.containsKey(sid)) {
-            return subscriptionIdToWebAppsOnWindowsMap.get(sid);
-        }
-        List<ResourceEx<WebApp>> webAppList = new ArrayList<>();
-        try {
-            Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
-            for (WebApp webApp : azure.webApps().list()) {
-                if (webApp.operatingSystem().equals(OperatingSystem.WINDOWS)) {
-                    webAppList.add(new ResourceEx<>(webApp, sid));
-                }
-            }
-            subscriptionIdToWebAppsOnWindowsMap.put(sid, webAppList);
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-        return webAppList;
+        return this.listWebAppsOnWindows(sid, force);
     }
 
     /**
-     * List all the Web Apps in selected subscriptions.
+     * List all the Web Apps on Windows in selected subscriptions.
      */
-    public List<ResourceEx<WebApp>> listAllWebAppsOnWindows(boolean force) {
-        List<ResourceEx<WebApp>> webAppList = new ArrayList<>();
-        List<Subscription> subscriptions = AzureMvpModel.getInstance().getSelectedSubscriptions();
-        for (Subscription sub : subscriptions) {
-            webAppList.addAll(this.listWebAppsOnWindowsBySubscriptionId(sub.subscriptionId(), force));
+    public List<ResourceEx<WebApp>> listAllWebAppsOnWindows(final boolean force) {
```

Comment:
`final` is good here. So can we apply this to other APIs if possible? It also makes them consistent and avoids confusion like "why some has final modifier and others not? Is there some tricks?"

Reply:
New added or updated function, I update them with ```final``` modifier, For those unchanged codes, I prefer not touch them for the moment. We could set an issue to track the code styling. https://github.com/Microsoft/azure-tools-for-java/issues/1960

Reply:
I don't think checkstyle can handle this. Whether a parameter needs `final` modifier should be determined by the meaning of function itself. Checkstyle can only detect whether you change value of a param in the body. 

I think for the obvious readonly parameters in this file, we can add `final` modifiers first.

Reply:
Updated for consistency, but still, keep the issue there to tack our style issue.